src/plugins/Wallet/nonce.ts

```diff
@@ -0,0 +1,49 @@
+import number from 'bignumber.js'
+import { web3 } from './web3'
+
+class NonceManager {
+    constructor(private address: string) {}
+    private nonce = NonceManager.INITIAL_NONCE
```

Comment:
what will happen if the Maskbook app gets closed? it seems like this info is stored in the memory currently

Reply:
Good point. I didn't test it yet. This nonce only fetching once when local cache absence and prefer cached nonce after that. If the user is going to commit new nonce in another place it may be staled the cache here.

Reply:
I made some changes to this service. For now, the nonce will update when we encounter a "nonce two low" error.

For some unknown reason, the cached nonce getting staled the user will be reported "nonce too low" with a donated failed dialog at that time the cached nonce will be reset to the remote nonce.